### PR TITLE
Enhance modularization of CtsCapitainsLocalResolver - Issue #154

### PR DESCRIPTION
**Pull request by PonteIneptique**
Dear @brosner ,
I'd like to get your feeling on this PR. I have a big issue with how Nautilus Resolver is inheriting but copy-pasting right now. I think this would allow to avoid redundancy of code later.

### CODE REVIEW

**Comment by PonteIneptique** [comment]
@balmas @MrGecko Can you have check on this branch ? We are moving to a dev branch system for the next version, as it might incorporate more than bugfixes.

**Comment by PonteIneptique** [comment]
There is another commit coming actually to make multiprocessing of parsing easier...

**Comment by PonteIneptique** [comment]
The last change will allow to change part of the parsing routine (like the dispatching) or/and change the glue method parse to do multiprocessing.

File: MyCapytain/resolvers/cts/local.py [hunk 1]
```diff
@@ -81,85 +109,125 @@ def xmlparse(self, file):
...
+                parent=textgroup,
+                _cls_dict=self.classes,
+                _with_children=True
+            )
+
+        return work, texts
+
+    def _parse_text(self, text, directory):
+        __textkey__, __text__ = text.id, text
+        # This if allows to avoid reparsing of a single file if it was already parsed
```
Comment (balmas) [inline MyCapytain/resolvers/cts/local.py]: not sure I understand this comment. Which if statement?

Comment (balmas) [inline MyCapytain/resolvers/cts/local.py]: not sure I understand this comment. Which if statement?

**Comment by brosner** [comment]
I like the direction this PR is taking. I wrote my own subclassed local resolver in Scaife mostly to better understand the resolver internals. This PR should actually eliminate the need I had which was mostly to make it more extensible.

The biggest pain point for us in Scaife Viewer is parsing the leaf node passages. For example, we need to get the text for `urn:cts:greekLit:tlg0016.tlg001.perseus-grc2:1.1.0`, `urn:cts:greekLit:tlg0016.tlg001.perseus-grc2:1.1.1` and so on... This is pretty expensive and I've been meaning to look into how this can be made faster.

I am digressing. This PR gets a +1 from me and look forward to trying it out on Scaife Viewer.

**Pull request merged**